dokka-subprojects/plugin-base/src/test/kotlin/expectActuals/ExpectActualsTest.kt

```diff
@@ -380,4 +381,58 @@ class ExpectActualsTest : BaseAbstractTest() {
             )
         }
     }
+
+
+    @Test
+    fun `should work with the reversed order of source sets #3798`() {
```

Comment:
would be nice to additionally have a test more similar to the issue #3798, where we do have an `expect` in shared source-set (e.g. `native`) and not in `common`

Also, it's not really obvious, why `reverse order` somehow affects the changes in PR for me...

Reply:
[DefaultDocumentableMerger.kt](https://github.com/Kotlin/dokka/pull/3875/files/ab76464f4ee06e320c68e57e74e8871e8b9ffbf6#diff-00add2e0f152afbda9fba4e4e7e8d666bcfb619525d9d73bb211e7776e7bd1a9) does not depend on a platform and not use it at all.


The configuration from the issue just affects the order of source sets. Intermediate source sets do not trigger the issue without the special order.
I have updated the description of the PR. Feel free to let me know if you have questions.